python/modelator/check.py

```diff
@@ -0,0 +1,157 @@
+import argparse
+import json
+import os
+from typing import Tuple, List
+
+from modelator_py.apalache.pure import apalache_pure
+from modelator_py.tlc.pure import tlc_pure
+from modelator_py.util.tlc import tlc_itf
+
+from .utils import apalache_helpers, tla_helpers, modelatorpy_helpers, tlc_helpers
+from .parse import parse
+from .typecheck import typecheck
+from typing import Dict
+from .utils.ErrorMessage import ErrorMessage
+from . import constants
+
+
+def check_tlc(
+    tla_file_name: str,
+    files: Dict[str, str],
+    args: Dict = None,
+    do_parse: bool = True,
+    config_file_name: str = None,
+) -> Tuple[bool, ErrorMessage, List]:
+
+    if do_parse is True:
+        parsable, msg = parse(tla_file_name=tla_file_name, files=files)
+        if parsable is False:
+            return (False, msg, [])
+
+    if config_file_name is not None:
+        if args is None:
+            args = {}
+        args["config"] = config_file_name
+
+    json_command = modelatorpy_helpers.wrap_command(
+        cmd=constants.CHECK_CMD,
+        checker=constants.TLC,
+        tla_file_name=tla_file_name,
+        files=files,
+        args=args,
+    )
+
+    result = tlc_pure(json=json_command)
+    if result["return_code"] == 0:
+        return (True, ErrorMessage(""), [])
+    else:
+        tlc_itf_json = {"stdout": result["stdout"], "lists": True, "records": True}
+
+        itf_trace_objects = tlc_itf(json=tlc_itf_json)
+
+        cex = itf_trace_objects[0]["states"]
```

Comment:
This needs to be changed. As we've seen in the online demo, there are cases when the returned list is empty.